orchagent/fgnhgorch.h

```diff
@@ -0,0 +1,114 @@
+#ifndef SWSS_FGNHGORCH_H
+#define SWSS_FGNHGORCH_H
+
+#include "orch.h"
+#include "observer.h"
+#include "intfsorch.h"
+#include "neighorch.h"
+
+#include "ipaddress.h"
+#include "ipaddresses.h"
+#include "ipprefix.h"
+#include "nexthopgroupkey.h"
+
+#include <map>
+
+/* Maximum next hop group number, TODO: enforcement of this */
+#define NHGRP_MAX_SIZE 128
+
+typedef uint32_t Bank;
+typedef std::set<NextHopKey> ActiveNextHops;
+typedef std::vector<sai_object_id_t> FGNextHopGroupMembers;
+typedef std::vector<uint32_t> HashBuckets;
+typedef std::map<NextHopKey, HashBuckets> FGNextHopGroupMap;
+typedef std::vector<FGNextHopGroupMap> BankFGNextHopGroupMap;
+typedef std::map<Bank,Bank> InactiveBankMapsToBank;
+
+struct FGNextHopGroupEntry
+{
+    sai_object_id_t         next_hop_group_id;      // next hop group id
+    int                     ref_count;              // reference count
```

Comment:
can it be less than 0 ? If not better to use uint32_t

Reply:
Thanks for the comment. Don't use ref_count in the code, so removed this.